client/components/ui/date-picker.js

```diff
@@ -24,8 +27,71 @@ export default class DatePicker extends React.Component {
         });
     }
 
+    setMaxorMin(pickerOptions, props) {
+        // Maybe a minimum or maximum value is set
+        pickerOptions.max = props.maxDate ? new Date(props.maxDate) : false;
+        pickerOptions.min = props.minDate ? new Date(props.minDate) : false;
+        return pickerOptions;
+    }
+
+    componentWillReceiveProps(newProps) {
+        let pickerOptions = this.setMaxorMin({}, newProps);
+        this.pickadate.set(pickerOptions);
+    }
+
+    localizationTable(prefix, suffix, tableToLocalize) {
+        let pref = prefix.length ? `${prefix}.` : '';
+        let suf = suffix.length ? `.${suffix}` : '';
+        return tableToLocalize.map(element => $t(`${pref}${element}${suf}`));
+    }
+
+    generateLocalizationObject() {
+        let monthTable = [
+            'january',
+            'february',
+            'march',
+            'april',
+            'may',
+            'june',
+            'july',
+            'august',
+            'september',
+            'october',
+            'november',
+            'december'
+        ];
+        let weekdaysTable = [
+            'sunday',
+            'monday',
+            'tuesday',
+            'wednesday',
+            'thursday',
+            'friday',
+            'saturday'
+        ];
+        return {
+            monthsFull: this.localizationTable('client.datepicker.monthsFull', '',
```

Comment:
As all calls give `client.datepicker.` as part of the prefix, and as `localizationTable` is in this particular class/file, maybe we can just set `client.datepicker.` within localizationTable, so that all calls here contain just the interesting part? `monthsFull`, `monthsShort` etc.
Plus, probably the call can fit on one line, then :-)